### PR TITLE
[PERF] Wan2.2 support RMSNorm NPU

### PR DESCRIPTION
**Pull request by fan2956**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
add norm layer
Wan2.2 use fused rmsnorm

## Test Plan
server
vllm serve wan2.2-i2v-diffusers
--omni
--port 8099
--usp 8
--use-hsdp
--vae-patch-parallel-size 8
--vae-use-tiling
--log-stats
--profiler-config '{"profiler": "torch", "torch_profiler_dir": "./vllm_profile"}'
curl
curl -X POST http://localhost:8099/v1/videos
-F "prompt=**********"
-F "input_reference=image"
-F "size=720x1280"
-F "fps=24"
-F "num_frames=81"
-F "guidance_scale=1.0"
-F "flow_shift=5.0"
-F "num_inference_steps=4"
-F "seed=42"

## Test Result
fused opt to rmsnorm
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [√ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [√ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [√ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/models/wan2_2/wan2_2_transformer.py [hunk 1]
```diff
@@ -376,8 +377,12 @@ def __init__(
         self.tp_inner_dim = self.num_heads * head_dim
 
         # QK normalization using vLLM's RMSNorm
-        self.norm_q = DistributedRMSNorm(self.tp_inner_dim, eps=eps)
-        self.norm_k = DistributedRMSNorm(self.tp_inner_dim, eps=eps)
+        if get_tensor_model_parallel_world_size() > 1:
+            self.norm_q = DistributedRMSNorm(self.tp_inner_dim, eps=eps)
+            self.norm_k = DistributedRMSNorm(self.tp_inner_dim, eps=eps)
```
Comment (gcanlin) [inline vllm_omni/diffusion/models/wan2_2/wan2_2_transformer.py:383]: Can't DistributedRMSNorm also benefit from the optimized kernels?

**Review by lishunyang12 — COMMENTED** [review]
## Review: [PERF] Wan2.2 support RMSNorm NPU

Thanks for the PR. The overall approach of introducing a `CustomOp`-based `LayerNorm` and `RMSNorm` in a shared `norm.py` layer module is reasonable. I have several concerns that should be addressed before merging.

### 1. MRO / double-init issue in `LayerNorm` (bug)

`LayerNorm` inherits from `nn.LayerNorm` **and** `CustomOp`. In `__init__`, you call `super().__init__(...)` which goes to `nn.LayerNorm.__init__`, but `CustomOp.__init__` (which normally sets `self._forward_method`) is **never called** via `super()` because `nn.LayerNorm.__init__` does not cooperatively call `super().__init__()`.

You work around this by manually calling `CustomOp.dispatch_forward(self)`, but this is fragile and breaks the invariant that `CustomOp.__init__` ran. A cleaner pattern:

```python
def __init__(self, dim, eps=1e-6, elementwise_affine=True):
    nn.LayerNorm.__init__(self, normalized_shape=dim, eps=eps, elementwise_affine=elementwise_affine)
    CustomOp.__init__(self)  # sets self._forward_method
```

This also removes the need to override `forward()` at all, since `CustomOp.forward` already dispatches via `self._forward_method`.

### 2. `RMSNorm.forward_npu` unconditionally imports `torch_npu` (crash risk)

```python
def forward_npu(self, x):
    import torch_npu
    output = torch_npu.npu_rms_norm(...)
```

Unlike `LayerNorm.forward_npu` which guards with `_HAS_MINDIESD` and falls back gracefully, `RMSNorm.forward_npu` will raise `ImportError` immediately if `torch_npu` is not installed. While `forward_npu` should only be dispatched on NPU platforms where `torch_npu` is available, adding a similar guard + fallback pattern would be more robust and consistent with how `LayerNorm` handles `mindiesd`.

### 3. Duplicated TP branching logic (maintainability)

The pattern:
```python
if get_tensor_model_parallel_world_size() > 1:
    self.norm_q = DistributedRMSNorm(...)
    self.norm_k = DistributedRMSNorm(...)
else:
    self.norm_q = RMSNorm(...)
    self.norm_k = RMSNorm(...)
```
is copy-pasted three times across `WanSelfAttention`, `WanCrossAttention`, and for `norm_added_k`. Consider extracting a helper function like `_make_rms_norm(hidden_size, eps)` that encapsulates this logic once.

### 4. Attribute name mismatch: `variance_epsilon` vs `eps`

`DistributedRMSNorm` stores its epsilon as `self.eps`, while the new `RMSNorm` stores it as `self.variance_epsilon`. Since both classes are used interchangeably (selected at runtime based on TP world size), this inconsistency could cause issues if any downstream code accesses the epsilon attribute by name. Please align the naming.

### 5. `FP32LayerNorm` import is now unused

After replacing `FP32LayerNorm` usages with `LayerNorm`, the import `from diffusers.models.normalization import FP32LayerNorm` in `wan2_2_transformer.py` should be removed if it is no longer referenced.

### 6. Minor: Removed `.float()` in cross-attention forward

```python
# Before:
norm_hidden_states = self.norm2(hidden_states.float()).type_as(hidden_states)
# After:
norm_hidden_states = self.norm2(hidden_states).type_as(hidden_states)
```

This is correct since `LayerNorm.forward_native` already casts to float internally. Just confirming this is intentional and tested -- the NPU path via `fast_layernorm` must also handle mixed precision correctly for this to be safe.

### Summary

The core idea is sound. The main issues are (1) the fragile MRO workaround in `LayerNorm.__init__`, (3) code duplication, and (4) the attribute name mismatch. Items 2 and 5 are minor cleanups. Please address these before merging.
